autotest/gdrivers/gpkg.py

```diff
@@ -4298,3 +4298,35 @@ def test_gpkg_gti_gpkg_ext(tmp_vsimem):
     ds = gdal.Open(filename)
     assert ds.GetDriver().ShortName == "GPKG"
     assert ds.GetRasterBand(1).Checksum() == 4672
+
+
+###############################################################################
+# Test rename a raster table with SQL
+
+
+def test_gpkg_rename_raster_table(tmp_vsimem):
+
+    test_layer_path = str(tmp_vsimem / "test_gpkg_rename_raster_table.gpkg")
+
+    src_ds = gdal.Open("data/small_world.tif")
+    ds = gdaltest.gpkg_dr.CreateCopy(
+        test_layer_path,
+        src_ds,
+        options=[
+            "TILE_FORMAT=PNG",
+            "RASTER_TABLE=foo",
+        ],
+    )
+    ds = None
+    src_ds = None
+
+    ds = gdal.OpenEx(test_layer_path, gdal.OF_RASTER | gdal.OF_UPDATE)
+    # Get layer name
+    layer_name = ds.GetMetadataItem("IDENTIFIER")
+    assert layer_name == "foo"
+    ds.ExecuteSQL("ALTER TABLE foo RENAME TO bar")
+    ds = None
+    ds = gdal.Open(test_layer_path, gdal.OF_RASTER)
+    layer_name = ds.GetMetadataItem("IDENTIFIER")
+    assert layer_name == "bar"
```

Comment:
I would also check the checksum is preserved after renaming, to check that everything went fine.

and it could be relevant to test for the absence of the "foo" string in the raw file (after vacuuming), as done in ogr_gpkg.py:3417-3425

Reply:
done